Fix expected help message to show header

madanalogy opened the pull request:
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [X] Bug Fix

<!--
    If this pull request is addressing an issue, link to the issue: "Fixes #xxx" or "Resolves #xxx"
-->
Resolves #644 

<!--
    Please ensure your pull request is ready:
    - Bug fix PR that is non-trivial **should** add a page or unit test for regression testing.
    - Feature PR **must** add a page to the user guide for demo.
    - Enhancement PR **should** update the user guide.

    Otherwise, prefix your PR title with "[WIP]".
-->

**What is the rationale for this request?**

Improves consistency of markbind's CLI interface.

**What changes did you make? (Give an overview)**

I added a function to handle help messages for use with commander. The function prints the banner and is used in commander's ```helpOption()``` method.

**Provide some example code that this change will affect:**
<!-- Paste the example code below: -->
```
function handleHelp(txt) {
  printHeader();
  logger.log('');
  return txt;
}
program
  .helpOption('-h, --help', handleHelp('output usage information'));
```

**Is there anything you'd like reviewers to focus on?**

Updating of the commander module to latest version, whether it was done properly.

**Testing instructions:**

From the main working directory, run ```node index.js -h```

**Proposed commit message: (wrap lines at 72 characters)**

Fixes CLI help command.

<!--
    See this link for more info on how to write a good commit message:
    https://oss-generic.github.io/process/docs/FormatsAndConventions.html#commit-message
-->

<!--
    Some of the responses that you gave to the previous questions might
    provide you with the information needed to craft your commit message.
-->

madanalogy commented:
Thank you for the information and suggestion :)

I have reverted the changes to the json files and implemented an override for outputHelp (instead of my earlier initial implementation). I will also be editing the PR title, thanks again for pointing me in the right direction!

yamgent commented on index.js:
```diff
@@ -37,11 +37,27 @@ function handleError(error) {
   process.exitCode = 1;
 }
 
+// The following code block overrides commander's outputHelp()
```
It would be better if the comment mentions the _why_ (the reason for overriding) as well.

You should mention that we want to customize the help message with MarkBind's header, but commander.js doesn't provide any API to customize the help message.

```js
// We want to customize the help message to print MarkBind's header,
// but commander.js does not provide an API directly for doing so.
// Hence we override commander's outputHelp() completely.
```

yamgent commented on index.js:
```diff
@@ -37,11 +37,27 @@ function handleError(error) {
+  process.stdout.write(cb(this.helpInformation()));
+  this.emit('--help');
+};
+module.exports = program;
```
Is this line needed? I don't think anybody outside `index.js` uses the `program` variable inside here.

yamgent commented on index.js:
```diff
@@ -37,11 +37,27 @@ function handleError(error) {
+      return passing;
+    };
+  }
+  process.stdout.write(cb(this.helpInformation()));
```
I was wondering whether it would be even better if we just shadow the original method, and utilise it in our new method. This way, we do not have to duplicate the original method's logic at all, in case there's an update to the commander's logic when we move to a new version:

```js
program.defaultOutputHelp = program.outputHelp;
program.outputHelp = function (cb) {
  printHeader();
  this.defaultOutputHelp(cb);
};
```

madanalogy commented:
Thank you for the suggestions! I have changed the implementation to shadow instead of replace and updated the code comment. I have also rebased and squashed my commits.
